index.bs

```diff
@@ -1229,6 +1229,29 @@ access to that data in a {{Window}} global.
 
 Issue: Define how this works.
 
+# User Contexts # {#user-contexts}
+
+A <dfn>user context</dfn> represents a collection of top-level navigables within a
+[=remote end=]. Each top-level navigable belongs to exactly one user context, and child
+navigables belong to the same user context as their parent. A [=user context=] without
+navigables belonging to it is an <dfn>empty user context</dfn>.
+
+A [=remote end=] always contains one [=user context=] referred as the <dfn>default user
+context</dfn>.
+
+Each user context represents an additional [=storage key=], so that no storage
+state is shared between different user contexts.
+
+User contexts other than the [=default user context=] have a <dfn>user context
+id</dfn>, which is a string set upon context creation. The [=default user
+context=]'s does not have a [=user context id=].
```

Comment:
What about `null` for default user context id?

Reply:
I think it might be easier to treat the default context id as undefined.

Reply:
Open to change that.